projects/sandbox/train/train/data_structures.py

```diff
@@ -133,3 +133,27 @@ def forward(self, X: torch.Tensor, y: torch.Tensor) -> torch.Tensor:
             # strain data with the sampled glitches
             X[mask, i] = glitches[:, 0]
         return X, y
+
+
+class SignalInverter(torch.nn.Module):
+    def __init__(self, prob: float = 0.5):
+        super().__init__()
+        self.prob = prob
+
+    def forward(self, X, y):
+        if self.training:
+            mask = torch.rand(size=X.shape[:-1]) < self.prob
```

Comment:
i.e., each IFO is flipped independently?

Reply:
Yeah exactly